ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/OptionsDropdown.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactNode } from 'react';
+import { Dropdown, DropdownToggle, Button, FocusTrap } from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+
+export type OptionsDropdownProps = {
+    children: ReactNode;
+    toggleId: string;
+};
+
+/**
+ * Component that uses a PatternFly `Dropdown` with the contents wrapped in a `FocusTrap` to
+ * enhance accessibility for interactive elements. Upon opening the dropdown, the browser will focus
+ * on the container's close button and only allow tab navigation within the container.
+ */
+function OptionsDropdown({ children, toggleId }: OptionsDropdownProps) {
+    const { isOpen, onToggle } = useSelectToggle();
+
+    function handleToggle(isDropdownOpen, event) {
+        if (event.key !== 'Tab') {
+            onToggle(isDropdownOpen);
+        }
+    }
+
+    return (
+        <Dropdown
+            className="pf-u-mr-sm"
+            toggle={
+                <DropdownToggle id={toggleId} toggleVariant="secondary" onToggle={handleToggle}>
+                    Options
+                </DropdownToggle>
+            }
+            position="right"
+            isOpen={isOpen}
+        >
+            <FocusTrap
+                focusTrapOptions={{
+                    // If there is no focusable element passed in `props.children`, this will put
+                    // the focus on the toggle button instead of throwing an error.
+                    fallbackFocus: `#${toggleId}`,
+                    clickOutsideDeactivates: true,
+                    escapeDeactivates: true,
+                }}
+            >
```

Comment:
Here is a thought for consideration instead of a request for change. 

What is pro and con of an alternative encapsulation via `DropdownFocusTrap` component to wrap the children of a `Dropdown` element?
* Potential pro: Is there a reuse case for children of other than `Dropdown` element?
* Potential con: If `handleToggle` function is also needed for it to work. On the other side of the coin, is the callback function relevant for accessibility in non-focus-trap use of `useSelectToggle` hook?

Reply:
The `handleToggle` is necessary only in this specific use case (focus-trapping a `Dropdown`), since PatternFly's default behavior will close the Dropdown when the user presses "tab". This is opposed to the behavior of a Popout, where "tab" iterates through the interactive items contained in the Popout body. Possibly another good argument to just convert these components to use `Popout` instead.